crypto/src/swap.rs

```diff
@@ -82,14 +231,124 @@ impl From<SwapPlaintext> for pb::SwapPlaintext {
     }
 }
 
+impl From<&SwapPlaintext> for [u8; SWAP_LEN_BYTES] {
+    fn from(swap: &SwapPlaintext) -> [u8; SWAP_LEN_BYTES] {
+        let mut bytes = [0u8; SWAP_LEN_BYTES];
+        bytes[0] = SWAP_TYPE;
+        bytes[1..65].copy_from_slice(&swap.trading_pair.to_bytes());
+        bytes[65..73].copy_from_slice(&swap.t1.to_le_bytes());
+        bytes[73..81].copy_from_slice(&swap.t2.to_le_bytes());
+        bytes[81..89].copy_from_slice(&swap.fee.0.to_le_bytes());
+        bytes[89..121].copy_from_slice(&swap.pk_d.0);
+        bytes[121..153].copy_from_slice(&swap.b_d.compress().0);
+        bytes
+    }
+}
+
+impl From<SwapPlaintext> for [u8; SWAP_LEN_BYTES] {
+    fn from(swap: SwapPlaintext) -> [u8; SWAP_LEN_BYTES] {
+        (&swap).into()
+    }
+}
+
+impl From<&SwapPlaintext> for Vec<u8> {
+    fn from(swap: &SwapPlaintext) -> Vec<u8> {
+        let mut bytes = vec![SWAP_TYPE];
+        bytes.extend_from_slice(&swap.trading_pair.to_bytes());
+        bytes.extend_from_slice(&swap.t1.to_le_bytes());
+        bytes.extend_from_slice(&swap.t2.to_le_bytes());
+        bytes.extend_from_slice(&swap.fee.0.to_le_bytes());
+        bytes.extend_from_slice(&swap.pk_d.0);
+        bytes.extend_from_slice(&swap.b_d.compress().0);
+        bytes
+    }
+}
```

Comment:
I think we could skip having this impl, to avoid having duplication with the array variant.